File: lib/base/src/main/java/com/blankj/base/BaseLazyFragment.java
package com.blankj.base;

import android.util.Log;

/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2018/03/23
 *     desc  : base about lazy fragment
 * </pre>
 */
public abstract class BaseLazyFragment extends BaseFragment {

    private static final String TAG = "BaseLazyFragment";

    private boolean isDataLoaded;

    public abstract void doLazyBusiness();

    @Override
    public void setUserVisibleHint(boolean isVisibleToUser) {
        Log.d(TAG, "setUserVisibleHint: " + isVisibleToUser);
        super.setUserVisibleHint(isVisibleToUser);
        if (isVisibleToUser && mContentView != null && !isDataLoaded) {
            doLazyBusiness();
            isDataLoaded = true;
        }
    }

    @Override
    public void doBusiness() {
        if (getUserVisibleHint() && !isDataLoaded) {
            doLazyBusiness();
            isDataLoaded = true;
        }
    }
}

File: lib/base/src/main/java/com/blankj/lib/base/BaseLazyFragment.kt
package com.blankj.lib.base

import android.util.Log

/**
 * ```
 * author: blankj
 * blog  : http://blankj.com
 * time  : 2018/03/23
 * desc  : base about lazy fragment
 * ```
 */
abstract class BaseLazyFragment : BaseFragment() {

    companion object {
        private const val TAG = "BaseLazyFragment"
    }

    private var isDataLoaded: Boolean = false

    abstract fun doLazyBusiness()

    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
        Log.d(TAG, "setUserVisibleHint: $isVisibleToUser")
        super.setUserVisibleHint(isVisibleToUser)
        if (isVisibleToUser && !isDataLoaded) {
            doLazyBusiness()
            isDataLoaded = true
        }
    }

    override fun doBusiness() {
        if (userVisibleHint && !isDataLoaded) {
            doLazyBusiness()
            isDataLoaded = true
        }
    }
}
